src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -357,15 +357,60 @@ class BezierCurve
           for(int k = 0; k <= end; ++k)
           {
             dCarray[k] = lerp(dCarray[k], dCarray[k + 1], t);
-            dWarray[k] = lerp(dWarray[k], dWarray[k + 1], t);
           }
         }
-        ptval[i] = dCarray[0] / dWarray[0];
+        ptval[i] = dCarray[0];
       }
 
       return ptval;
     }
-    else  // If ordinary Bezier curve
+    // If rational, construct 4D homogeneous curve to evaluate
+    else
+    {
+      // Store BezierCurve of projective weights, (wx, wy, wz)
+      //  and BezierCurve of weights (w)
+      BezierCurve<T, NDIMS> projective(ord);
+      BezierCurve<T, 1> weights(ord);
+
+      for(int p = 0; p <= ord; ++p)
+      {
+        weights[p][0] = m_weights[p];
+
+        for(int i = 0; i < NDIMS; ++i)
+        {
+          projective[p][i] = m_controlPoints[p][i] * m_weights[p];
+        }
+      }
```

Comment:
Does this new refactor have approximately the same cost as the previous one?

Reply:
It should be an pretty significantly improved cost, full stop. Previously, the `lerp`ing of the weights would be done alongside the coordinates, even though the weights didn't change between dimension like the coordinates did. This method avoids that completely, so we're going from 6 de Casteljau loops to the minimum number of 4.